test/cmd/api.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/bash
+
+#
+# Test osbuild-composer's main API endpoint by building a sample image and
+# uploading it to AWS.
+#
+# This script sets `-x` and is meant to always be run like that. This is
+# simpler than adding extensive error reporting, which would make this script
+# considerably more complex. Also, the full trace this produces is very useful
+# for the primary audience: developers of osbuild-composer looking at the log
+# from a run on a remote continuous integration system.
+#
+
+set -euxo pipefail
+
+
+#
+# Verify that this script is running in the right environment. In particular,
+# it needs variables is set to access AWS.
+#
+
+printenv AWS_REGION AWS_BUCKET AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY > /dev/null
+
+
+#
+# Create a temporary directory and ensure it gets deleted when this script
+# terminates in any way.
+#
+
+WORKDIR=$(mktemp -d)
+function cleanup() {
+  rm -rf "$WORKDIR"
+}
+trap cleanup EXIT
+
+
+#
+# Install the aws client from the upstream release, because it doesn't seem to
+# be available as a RHEL package.
+#
+
+if ! hash aws; then
```

Comment:
What about awscli from EPEL?

Reply:
Good point! I've taken this verbatim from `aws.sh`. Do you mind if we do this in a follow-up?

Filed https://github.com/osbuild/osbuild-composer/issues/1037 to not forget.